#! /usr/bin/env python

PACKAGE='ptam'
import roslib; roslib.load_manifest(PACKAGE)

#from driver_base.msg import SensorLevels
from dynamic_reconfigure.parameter_generator import *

gen = ParameterGenerator()

# @todo Double check these ranges, defaults

RESTART_AR     = gen.const("SEND_PARAMETERS",          int_t, 0x00000001, "send_parameters")


#       Name                         Type      Reconfiguration lvl    Description                                                   Default             Min     Max
gen.add("ShowPC",                    bool_t,    0,              "Show point cloud",                                                 False)
gen.add("ShowAllKFs",                bool_t,    0,              "Show all keyframes ever made",                                     False)
gen.add("ShowKFs",                   bool_t,    0,              "Show keyframes",                                                   False)
gen.add("KFLifetime",                int_t,     0,              "KF lifetime in sec",                                               1,                   1,     60000)
gen.add("ShowPath",                  bool_t,    0,              "Show path",                                                        False)
gen.add("PathLength",                int_t,     0,              "Number of path anchor points",                                     1000,                1,     60000)
gen.add("ExportPC",                  bool_t,    0,              "export point cloud to file",                                       False)
gen.add("ExportPrefix",              str_t,     0,              "prefix for export point cloud to file",                            "")
gen.add("SaveMap",                   bool_t,    0,              "save full map to file",                                            False)
gen.add("KFFlags",                   int_t,     0,              "0:get all KFs, <0: get N newest KFs, 1: get new KFs only",         0,                  -100,   1)


exit(gen.generate(PACKAGE, "Config", "PTAMVisualizerParams"))
